src/infra/src/table/migration/m20250205_000001_create_reports_table.rs

```diff
@@ -0,0 +1,346 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+const REPORTS_FOLDERS_FK: &str = "reports_folders_fk";
+const REPORTS_FOLDER_ID_IDX: &str = "reports_folder_id_idx";
+
+const REPORT_DASHBOARDS_REPORTS_FK: &str = "report_dashboards_reports_fk";
+const REPORT_DASHBOARDS_DASHBOARDS_FK: &str = "report_dashboards_dashboards_fk";
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(create_reports_table_statement())
+            .await?;
+        manager
+            .create_index(create_reports_folder_id_idx_stmnt())
+            .await?;
+        manager
+            .create_table(create_report_dashboards_table_statement())
+            .await?;
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_table(Table::drop().table(ReportDashboards::Table).to_owned())
+            .await?;
+        manager
+            .drop_index(
+                Index::drop()
+                    .name(REPORTS_FOLDER_ID_IDX)
+                    .table(Reports::Table)
+                    .to_owned(),
+            )
+            .await?;
+        manager
+            .drop_table(Table::drop().table(Reports::Table).to_owned())
+            .await?;
+        Ok(())
+    }
+}
+
+/// Statement to create the reports table.
+fn create_reports_table_statement() -> TableCreateStatement {
+    Table::create()
+        .table(Reports::Table)
+        .if_not_exists()
+        // The ID is 27-character human readable KSUID.
+        .col(
+            ColumnDef::new(Reports::Id)
+                .char_len(27)
+                .not_null()
+                .primary_key(),
+        )
+        // Foreign key to the folders table. This is a 27-character human
+        // readable KSUID.
+        .col(ColumnDef::new(Reports::FolderId)
+            .char_len(27)
+            .not_null())
+        .col(ColumnDef::new(Reports::Name).string_len(256).not_null())
+        .col(ColumnDef::new(Reports::Title).string_len(256).not_null())
+        .col(ColumnDef::new(Reports::Description).text().null())
+        .col(ColumnDef::new(Reports::Enabled).boolean().not_null())
+        .col(ColumnDef::new(Reports::Frequency).json().not_null())
+        .col(ColumnDef::new(Reports::Destinations).json().not_null())        
+        .col(ColumnDef::new(Reports::Message).text().null())
```

Comment:
Seaorm uses `TEXT` for both postgres and mysql, and mysql `TEXT` data size is very less (64KB) compared to that of postgres. But ideally that should be enough even in case of mysql.